src/java.base/share/classes/java/security/ProtectionDomain.java

```diff
@@ -69,7 +71,9 @@ public class ProtectionDomain {
             "true".equals(GetPropertyAction.privilegedGetProperty(
                 "jdk.security.filePermCompat"));
 
-    private static class JavaSecurityAccessImpl implements JavaSecurityAccess {
+    static class JavaSecurityAccessImpl implements JavaSecurityAccess {
+        /* cache a copy for recording purposes */
+        static Properties initialSecurityProperties;
```

Comment:
This doesn't look very clean. Could the Security class hold the initial security properties and provide an accessor method that JavaSecurityAccess:getIinitialProperties could use?

Reply:
Agree, and alternatively, it seems cleaner to add a new SharedSecrets class for `java.security.Security` and remove the dependency on PD.

Reply:
modified code to have Security class hold the initial properties and provided an accessor method

Reply:
What about creating a new `JavaSecurityPropertiesAccess` class and moving the accessor method there? It seems it would be cleaner to remove the dependency on PD in the long run.

Reply:
@seanjmullan  - I looked at that approach. The `SharedSecrets.getJavaSecurityAccess().getInitialProperties();` call may trigger early initialization of the `java.security.Security` class - I'm not sure if we want that. ProtectionDomain class is currently loaded early in the JDK boot cycle. 

In fact the change suggested by @AlanBateman yesterday also has possibility to trigger unnecessary loading of the Security class. I might revert to the original design where we store the cached Properties in ProtectionDomain ?

Reply:
Maybe I am missing something. If this JFR event is enabled, and the properties have not yet been accessed, then it seems ok for JFR to load the `Security` class when JFR is started since the user is interested in this event.

Reply:
My own thoughts here would be that the JFR event system should try and avoid side effects in such areas. If the Security class hasn't been loaded at time of recording, then I'd argue that the number InitialSecurityProperty events should be zero. (which is not possible at the moment since JFR does trigger Security class loading itself anyway)

Reply:
After further conversation with @seanjmullan , a sharedSecrets accessor for the Properties map should be ok. Edits pushed.